### PR TITLE
Pair/feature/api calls

### PR DESCRIPTION
**Pull request by Speekins**
#### What does this PR do?
- This merge will add fetch API call to get data from server and display on page load.
  - API call located in file `apiCalls.js` 
- Also adds error handling for level 400 and 500 status response codes.

#### What (if any) features are you implementing?
- `Fetch API`
- `throw`
- `Error()`

#### What (if anything) did you refactor?
N/A

#### Were there any issues that arose?
N/A

#### Any other comments, questions, or concerns?
None.

#### Screenshots (if applicable)
image
image
image

### CODE REVIEW

**Comment by bkuhn2** [comment]
Looks good - per the project spec this successfully fetches the data and and has both DOM and behind the scenes user handling.

File: src/apiCalls.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+function fetchData(url) {
```
Comment (sertmer) [inline src/apiCalls.js:1]: Can you think of a more semantic or descriptive name? What kind of data?

File: src/apiCalls.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  return fetch(url)
+  .then(response => {
+    if (response.ok) {
+      return response.json()
+    } else if (response.status >= 500) {
+      throw new Error('The server\'s messed up!')
+    } else if (response.status >= 400 && response.status < 500) {
+      throw new Error('This is a level 400 error')
```
Comment (sertmer) [inline src/apiCalls.js:9]: Glad to see you thinking of those errors and what you'd wanna communicate to the user

File: src/components/App/App.js [hunk 3]
```diff
@@ -1,24 +1,30 @@
...
+    .then((data) => this.setState({ allMovies: data.movies }))
+    .catch(error => this.setState({ error: error}))
   }
 
   render(){
     return (
       <div>
         <h1>test</h1>
+        {console.log(this.state.error)}
+        {this.state.error && <h2>Something went wrong! {this.state.error.message}</h2>}
```
Comment (sertmer) [inline src/components/App/App.js:27]: Great use of that `logical and` operator for that conditional rendering

**Review by sertmer — COMMENTED** [review]
I've left a few comments - but overall this looks great! Let me know if you have any questions!

**Pull request merged**